### PR TITLE
Provide useful defaults for Android logcat output

### LINKED ISSUE #21637: println doesn't appear in logcat output
This makes it harder to interact with code that is already built around println, like various `-Z` options. Is there some way to redirect stdout to logcat, @paulrouget? Or perhaps @fabricedesre knows?

### LINKED ISSUE #21783: Panic messages don't appear in logcat
This means that assertions, unwrap, etc. don't leave any trace. The log shows a sigill, and I can only figure out what's happening if I run the app under gdb. I wonder if we could add a call to `__android_log_write` in the global panic hook to make this work.

### PR DESCRIPTION
**Pull request by jdm**
These changes integrate the stdout/stderr redirection from https://github.com/tomaka/android-rs-glue/blob/0e80cfbcee362f207389dc2660525c3d74987f70/cargo-apk/injected-glue/lib.rs#L240-L303, and set up the default RUST_LOG filters so that useful JS errors and GL errors appear in the logcat output. I have verified that thread panics appear in output as well (without backtraces, sadly) by visiting https://acid3.acidtests.org.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21637 and fix #21783
- [x] These changes do not require tests because no tests for logcat.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21812)

### CODE REVIEW

**Comment by SimonSapin** [comment]
Sorry I lagged a bit, I’m about to submit a review with some code changes.

**Review by SimonSapin — CHANGES_REQUESTED** [review]
For backtraces, maybe `std::env::set_var("RUST_BACKTRACE", "1")` would help?

**Comment by jdm** [comment]
Setting RUST_BACKTRACE=1 tends to hit segfaults in libunwind when using NDK12. I'll give it a try as part of the NDK15 upgrade.

File: ports/libsimpleservo/src/jniapi.rs [hunk 1]
```diff
@@ -434,3 +446,92 @@ fn initialize_android_glue(env: &JNIEnv, activity: JObject) {
...
+                            buf[cursor] = b'\0' as c_char;
+                            buf.set_len(cursor + 1);
+                        } else {
+                            cursor = 0;
+                        }
+                    } else {
+                        cursor = buf.len();
+                    }
+                    if cursor == buf.capacity() - 1 {
+                        __android_log_write(3, tag, buf.as_ptr());
```
[the remark this answers is not in the document]
Comment (jdm) [reply ports/libsimpleservo/src/jniapi.rs]: https://github.com/tomaka/android-rs-glue/issues/183

File: ports/libsimpleservo/src/jniapi.rs [hunk 2]
```diff
@@ -434,3 +446,92 @@ fn initialize_android_glue(env: &JNIEnv, activity: JObject) {
...
+                        buf.set_len(0);
+                        cursor = 0;
+                    }
+                }
+            }
+        }
+
+        let mut thread = mem::uninitialized();
+        let result = pthread_create(&mut thread, ptr::null(), logging_thread,
```
[the remark this answers is not in the document]
Comment (jdm) [reply ports/libsimpleservo/src/jniapi.rs]: I tested using the stdlib threads and it works fine on my Pixel 2.

**Comment by jdm** [comment]
I used your code changes unmodified.

**Review by SimonSapin — APPROVED** [review]

**Pull request merged**